crates/extensions/c8y_mapper_ext/src/tests.rs

```diff
@@ -1707,114 +1541,78 @@ async fn mapper_dynamically_updates_supported_operations_for_child_device() {
+            &Topic::new_unchecked("te/device/child11//"),
+            json!({
+                "@type":"child-device",
+                "@id": "child11",
+                "name": "child11",
+                "@parent": "device/child1//"
+            })
+            .to_string(),
+        )
+        .with_retain(),
     )
-    .await;
-}
-
-#[tokio::test]
-async fn mapping_dynamically_added_child_device_dir_with_default_external_id_naming_scheme() {
-    let cfg_dir = TempTedgeDir::new();
-    let (mqtt, _http, mut fs, _timer, _ul, _dl) = spawn_c8y_mapper_actor(&cfg_dir, false).await;
-    let mut mqtt = mqtt.with_timeout(TEST_TIMEOUT_MS);
-    skip_init_messages(&mut mqtt).await;
-    let ops_dir = cfg_dir.dir("operations").dir("c8y");
-
-    // Simulate FsEvent for the creation of a new child device following default naming scheme
-    fs.send(FsWatchEvent::DirectoryCreated(
-        ops_dir.dir("test-device:device:child").to_path_buf(),
-    ))
     .await
     .expect("Send failed");
 
-    // Expect auto-registration message
-    assert_received_includes_json(
-        &mut mqtt,
-        [(
-            "te/device/child//",
-            json!({"@type":"child-device", "@id": "test-device:device:child", "name": "child"}),
-        )],
-    )
-    .await;
-
-    // ...and the corresponding SmartREST registration message
     assert_received_contains_str(
         &mut mqtt,
-        [(
-            "c8y/s/us",
-            "101,test-device:device:child,child,thin-edge.io-child",
-        )],
+        [
+            ("c8y/s/us", "101,child1,child1,thin-edge.io-child"),
+            ("c8y/s/us/child1", "101,child11,child11,thin-edge.io-child"),
+        ],
     )
     .await;
-}
-
-#[tokio::test]
-async fn mapping_dynamically_added_child_device_dir_with_forbidden_characters() {
-    let cfg_dir = TempTedgeDir::new();
-    let ops_dir = cfg_dir.dir("operations").dir("c8y");
-    let (mqtt, _http, mut fs, _timer, _ul, _dl) = spawn_c8y_mapper_actor(&cfg_dir, false).await;
-    let mut mqtt = mqtt.with_timeout(TEST_TIMEOUT_MS);
-    skip_init_messages(&mut mqtt).await;
-
-    // Simulate FsEvent for the creation of a new child device with a dir name containing forbidden chars
-    let child_dir = ops_dir.dir("my#complex+child");
-    fs.send(FsWatchEvent::DirectoryCreated(child_dir.to_path_buf()))
-        .await
-        .expect("Send failed");
-
-    // No mapped messages as my#complex+child is ignored
-    assert!(mqtt.recv().await.is_none());
 
-    // Validate that further directory creation events are still processed
-    fs.send(FsWatchEvent::DirectoryCreated(
-        ops_dir.dir("simple_child").to_path_buf(),
-    ))
+    // Send any command metadata message to trigger the supported operations update
+    mqtt.send(
+        MqttMessage::new(
+            &Topic::new_unchecked("te/device/child11///cmd/c8y_ChildTestOp3"),
+            "{}",
+        )
+        .with_retain(),
+    )
     .await
     .expect("Send failed");
 
```

Comment:
Maybe add an assertion for the file creation of `c8y_ChildTestOp3`?

Reply:
Since the file creation is more of an internal impl detail and the next assertion is asserting the effect via the public API anyway, I think we can skip it.